.eslintrc

```diff
@@ -1,11 +1,21 @@
 {
-    "extends": "dhis2",
+    "extends": [
+        "react-app",
+        "prettier"
+    ],
+    "settings": {
+        "react": {
+            "version": "^15.3.1"
+        }
+    },
     "env": {
         "browser": true,
         "jest": true
     },
     "rules": {
-        "no-param-reassign": [2, { "props": false }],
+        "no-param-reassign": "off",
```

Comment:
I think it might be unwise to have this completely off. As JS is pass by value, this rule catches the cases where parameters are directly being reassigned instead of nested properties. 

Only reassignment of nested properties should allowed.

Reply:
I tried switching the rule back on as we had it. This resulted in three errors. In all these cases, I would say that parameter re-assignment was a valid approach to solve the problem in question. Please review these and see what you think:
- https://github.com/dhis2/maintenance-app/blob/TECH-93/tweak-eslint-configuration/src/utils/LoadableComponent.js#L61-L63
- https://github.com/dhis2/maintenance-app/blob/TECH-93/tweak-eslint-configuration/src/EditModel/SingleModelStore.js#L150
- https://github.com/dhis2/maintenance-app/blob/TECH-93/tweak-eslint-configuration/src/EditModel/formHelpers.js#L195

I kind of agree with you that parameter re-assignment in general is a pretty bad idea. Perhaps what we could do, is just keep the rule as it was, and place ESLint disabling comments around the three parts of code listed above.

Thanks for the feedback. Looking forward to getting more of it 🔢

Reply:
Yeah. For the two first cases i do agree that the reassignment serves a purpose, at least in the way the code is constructed. The reassignment in the third case seems redundant as it's already handled in the parameters. 

I think parameter reassignment may often be an indicator that the code structure might be a bit off, and having the waring on in those cases requires the coder to consider the solution one more time. When adding ESlint disabling at least you have to make a conscious decision.